services/users.js

```diff
@@ -21,7 +21,14 @@ async function signup (firstName, lastName, email) {
 }
 
 async function login (secret) {
-  const userData = await UserModel.getUserBySecret(secret)
+  let userData
+  const trueUserSecret = await UserModel.getTrueUserSecretBySecret(secret)
```

Comment:
Conceptually, the `UserModel.getTrueUserSecretBySecret` should just return back `secret` (rather than null) if `mergedUserId` doesn't exist on the account.  

Currently, if it returns null, that could mean that either the user doesn't have a merged account OR `secret` is invalid. If we changed it so it always returns a valid secret if it exists, we wouldn't need the if-statement below. 

```
  const trueUserSecret = await UserModel.getTrueUserSecretBySecret(secret)
  if (trueUserSecret == null) {
    userData = await UserModel.getUserBySecret(secret)
  } else if (trueUserSecret != null) {
    userData = await UserModel.getUserBySecret(trueUserSecret)
  }

if (userData == null) {
    throw new Error('invalidSecret')
  }
```
would become

```
 // trueUserSecret === secret if user doesn't have merged account
  const trueUserSecret = await UserModel.getTrueUserSecretBySecret(secret)
  if (trueUserSecret == null) {
    throw new Error('invalidSecret')
  }
  await UserModel.getUserBySecret(trueUserSecret)
```

Reply:
Absolute genius